lib/screens/shopping_lists.dart

```diff
@@ -16,43 +17,72 @@ class _HomePageState extends State<HomePage> {
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        title: const Text('Choose your list!'),
-      ),
-      body: FutureBuilder<List<ShoppingList>>(
-          initialData: const [],
-          future: dbController.getShoppingLists(),
-          builder: (context, snapshot) {
-            if (!snapshot.hasData) {
-              return const Center(child: CircularProgressIndicator());
-            } else {
-              return ListView.builder(
-                padding: const EdgeInsets.all(16.0),
-                itemCount: snapshot.data!.length,
-                itemBuilder: (context, index) {
-                  return MyTile(snapshot.data![index].title,
-                      snapshot.data![index].id!, _deleteNote,
-                      key: UniqueKey());
-                },
-              );
+    return BlocProvider(
+      create: (bloc_context) => ShoppingListCubit(),
+      child: Scaffold(
+        appBar: AppBar(
+          title: const Text('Choose your list!'),
+        ),
+        body: BlocConsumer<ShoppingListCubit, ShoppingListState>(
+          listener: (context, state) {
+            print(state.dataState);
+          },
+          builder: (context, state) {
+            BlocProvider.of<ShoppingListCubit>(context).getData();
+            switch (state.dataState) {
+              case DataProcessingState.fetchingData:
+                {
+                  return const Center(child: CircularProgressIndicator());
+                }
+              case DataProcessingState.dataAccessed:
+                {
+                  return ListView.builder(
+                    padding: const EdgeInsets.all(16.0),
+                    itemCount: BlocProvider.of<ShoppingListCubit>(context)
+                        .ShoppingListData
+                        .length,
```

Comment:
to powinno być wczytywane ze state'a a nie z cubita
```suggestion
                    itemCount: state.ShoppingListData.length,
```